### PR TITLE
out.d and machobj.d: replace static if with if

### PR DESCRIPTION
**Pull request by WalterBright**

### CODE REVIEW

**Comment by thewilsonator** [comment]
indentation

**Comment by WalterBright** [comment]
> indentation

I deliberately did not adjust that in order to make the review much easier. A separate PR for whitespace-only changes would be apropos.

File: src/dmd/backend/cgobj.d [hunk 1]
```diff
@@ -104,7 +104,7 @@ int obj_namestring(char *p,const(char)* name);
 version (MARS)
 {
 // C++ name mangling is handled by front end
-const(char)* cpp_mangle(Symbol* s) { return s.Sident.ptr; }
+const(char)* cpp_mangle(Symbol* s) { return &s.Sident[0]; }
```
Comment (thewilsonator) [inline src/dmd/backend/cgobj.d:107]: seems unrelated

File: src/dmd/backend/out.d [hunk 2]
```diff
@@ -459,7 +459,7 @@ void dt_writeToObj(Obj objmod, dt_t *dt, int seg, ref targ_size_t offset)
                     objmod.reftocodeseg(seg,offset,dt.DTabytes);
                 else
                 {
-static if (TARGET_LINUX || TARGET_OSX || TARGET_FREEBSD || TARGET_OPENBSD || TARGET_DRAGONFLYBSD || TARGET_SOLARIS)
+if (config.exe & EX_posix)
```
Comment (thewilsonator) [inline src/dmd/backend/out.d:462]: indentation

File: src/dmd/backend/cgobj.d (hunk 1 above)
Comment (WalterBright) [reply src/dmd/backend/cgobj.d:107]: It is unrelated, but it matches what I have in my other version of the file.

File: src/dmd/backend/out.d [hunk 3]
```diff
@@ -1417,7 +1417,7 @@ version (SCPP)
 {
                 objmod.codeseg(cpp_mangle(funcsym_p),1);
 }
-else static if (TARGET_WINDOS)
+else if (config.exe & EX_windos)
```
Comment (thewilsonator) [inline src/dmd/backend/out.d:1420]: `version () {} else if` is not the easiest to understand at a glance.

Comment (WalterBright) [reply src/dmd/backend/out.d:1420]: Yeah, but it doesn't mess up all the indentation. See my other remarks on indentation.

**Comment by Geod24** [comment]
> I deliberately did not adjust that in order to make the review much easier. A separate PR for whitespace-only changes would be apropos.

Such trick is not needed. One can just hide whitespace change and get the same diff. Most reviewers know about this already.

Screen Shot 2020-11-13 at 11 10 21

**Comment by WalterBright** [comment]
I do know about this, @Geod24. But the differences are still there, as `git diff` will show, or any other file comparison. I like to have whitespace diffs in their own PR. That way, I can ignore that one when looking for the source of problems.

**Comment by Geod24** [comment]
`git diff`:
```
--ignore-all-space     -- ignore white space when comparing lines
--ignore-blank-lines   -- do not show hunks that add or remove blank lines
--ignore-cr-at-eol     -- ignore carriage-return at end of line
--ignore-space-at-eol  -- ignore changes in whitespace at end of line
--ignore-space-change  -- ignore changes in amount of white space
--ignore-submodules    -- ignore changes to submodules
```

If your diff tool does not support ignoring whitespace, then pick a new one.

**Comment by WalterBright** [comment]
What I did was change `static if` to `if`. Suddenly, the formatting becomes a problem? Shouldn't PRs be doing one problem at a time?

**Comment by Geod24** [comment]
Because the indentation was already broken, and it is broken because a previous PR left it broken under a similar argument that it will eventually be fixed.

**Comment by WalterBright** [comment]
> Because the indentation was already broken,

You could say "why didn't you fix this or that while you're at it" to about everything else in those files.

I practice what I preach, which is when fixing a problem fix the problem and nothing else in a PR. Fix other problems in other PRs. It's pretty frustrating when faced with a PR that does several things right and several things wrong, whatcha gonna revert? And have you ever noticed that the bigger a PR is the more insipid the reviews get, if any?

> under a similar argument that it will eventually be fixed.

This PR _is_ fixing technical debt.

**Comment by Geod24** [comment]
> You could say "why didn't you fix this or that while you're at it" to about everything else in those files.

Don't get me wrong, I didn't mean to block this PR. I was just argumenting that leaving off indentation fixes in a PR is an outdated practice that needs to go. However, when I raised that point, I didn't see that the indentation was already broken.
So let's merge it and let the whitespace diff for another day.

**Review by Geod24 — APPROVED** [review]

**Comment by WalterBright** [comment]
Thank you, @Geod24 

Can you please restart buildkite while you're at it? I think it's a heisenbug, but need confirmation.

**Comment by Geod24** [comment]
Yep, I just did. You should be able to, too.

**Comment by WalterBright** [comment]
> You should be able to, too.

I used to be able to. But that disappeared a couple months ago.

**Comment by thewilsonator** [comment]
Failure seems persistent, try once pushing. 
cc @9il

**Comment by Geod24** [comment]
> I used to be able to. But that disappeared a couple months ago.

I checked and you are still part of the team. Are you logged in with your digitalmars address ?

**Comment by Geod24** [comment]
Yes, the failure is on mir side. AFAICS they are aware of the problem, there were two new patch releases pushed  recently. Re-triggered again, hopefully it'll go green.

**Comment by 9il** [comment]
> Yes, the failure is on mir side. AFAICS they are aware of the problem, there were two new patch releases pushed recently. Re-triggered again, hopefully it'll go green.

The releases were for `mir-algorithm`. Let me check how it works with `mir`

**Comment by Geod24** [comment]
Last successful build on master:
```

~> dub test --compiler=dmd
--
  | Fetching mir-core 1.1.14 (getting selected version)...
  | Fetching mir-random 2.2.15 (getting selected version)...
  | Fetching mir-linux-kernel 1.1.0 (getting selected version)...
  | Fetching mir-algorithm 3.9.47 (getting selected version)...
  | Generating test runner configuration 'mir-test-unittest' for 'unittest' (sourceLibrary).
  | Excluding package.d file from test due to https://issues.dlang.org/show_bug.cgi?id=11847
  | Excluding package.d file from test due to https://issues.dlang.org/show_bug.cgi?id=11847
  | Excluding package.d file from test due to https://issues.dlang.org/show_bug.cgi?id=11847
  | Performing "unittest" build using dmd for x86_64.
  | mir-core 1.1.14: building configuration "library"...
  | mir-algorithm 3.9.47: building configuration "default"...
  | mir-random 2.2.15: building configuration "extended"...
  | mir 3.2.1: building configuration "mir-test-unittest"...
  | Linking...
  | Running ./mir-test-unittest
```

Current failing build:
```

~> dub test --compiler=dmd
--
  | Fetching mir-core 1.1.16 (getting selected version)...
  | Fetching mir-random 2.2.15 (getting selected version)...
  | Fetching mir-linux-kernel 1.1.0 (getting selected version)...
  | Fetching mir-algorithm 3.10.2 (getting selected version)...
  | Generating test runner configuration 'mir-test-unittest' for 'unittest' (sourceLibrary).
  | Excluding package.d file from test due to https://issues.dlang.org/show_bug.cgi?id=11847
  | Excluding package.d file from test due to https://issues.dlang.org/show_bug.cgi?id=11847
  | Excluding package.d file from test due to https://issues.dlang.org/show_bug.cgi?id=11847
  | Performing "unittest" build using dmd for x86_64.
  | mir-core 1.1.16: building configuration "library"...
  | mir-algorithm 3.10.2: building configuration "default"...
  | mir-random 2.2.15: building configuration "extended"...
  | mir 3.2.1: building configuration "mir-test-unittest"...
  | Linking...
  | Running ./mir-test-unittest
  | source/mir/sparse/package.d(117): [unittest] unittest failure
  | core.exception.AssertError@source/mir/sparse/package.d(117): unittest failure
```

`mir-algorithm` and `mir-core` have been updated.
I just triggered a build on master just to be sure (https://buildkite.com/dlang/dmd/builds/17138).

**Comment by 9il** [comment]
Yes, this is mir failure, confirmed.

**Comment by Geod24** [comment]
@9il : The build on master failed, so it is indeed a mir problem. Merging this in the meantime.
